lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -498,7 +545,32 @@ void TruncatedDistribution::setFiniteUpperBound(const Bool finiteUpperBound)
 
 Bool TruncatedDistribution::getFiniteUpperBound() const
 {
-  return finiteUpperBound_;
+  LOGWARN("TruncatedDistribution::getFiniteUpperBound is deprecated, use getRange.");
+  return bounds_.getFiniteUpperBound()[0];
+}
+
+void TruncatedDistribution::setBounds(const Interval & bounds)
+{
+  if (bounds_ != bounds)
+  {
+    const Scalar cdfLowerBound = distribution_.computeCDF(bounds.getLowerBound());
+    const Scalar cdfUpperBound = distribution_.computeCDF(bounds.getUpperBound());
+    if (cdfUpperBound_ <= cdfLowerBound_) throw InvalidArgumentException(HERE) << "Error: the truncation interval does not contain a non-empty part of the support of the distribution";
+    bounds_ = bounds;
+    pdfLowerBound_ = distribution_.computePDF(bounds.getLowerBound());
+    pdfUpperBound_ = distribution_.computePDF(bounds.getUpperBound());
+    cdfLowerBound_ = cdfLowerBound;
+    cdfLowerBound_ = cdfUpperBound;
+    isAlreadyComputedMean_ = false;
+    isAlreadyComputedCovariance_ = false;
+    isAlreadyCreatedGeneratingFunction_ = false;
+    computeRange();
+  }
+}
+
+Interval TruncatedDistribution::getBounds() const
+{
+  return bounds_;
```

Comment:
bounds_ or range_?

Reply:
I keep bounds_ as the bounds provided by the user.